packages/babel-core/src/transformation/normalize-opts.ts

```diff
@@ -48,10 +55,11 @@ export default function normalizeOptions(config: ResolvedConfig) {
       minified: opts.minified,
 
       // Source-map generation flags.
-      sourceMaps,
-
+      // babel-generator does not differentiate between `true`, `"inline"` or `"both"`
+      sourceMaps: !!sourceMaps,
```

Comment:
Here we cast `generatorOpts.sourceMaps` to boolean as required by the `@babel/generator`.